### PR TITLE
refactor(eval): replace model@format strings with typed ModelConfig dataclass

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

Replaces the fragile `model@format` string concatenation pattern in the eval system with a proper typed `ModelConfig` dataclass. This is the deeper internal refactor requested in https://github.com/gptme/gptme/pull/1271#issuecomment-3916479388 — making tool format a "separate column/tuple-thing" rather than encoding it via `@` in the model name.

**Supersedes PR #1272** which only separated format in CSV display. This PR does the full internal refactor.

### Changes

- **`types.py`**: New frozen `ModelConfig(model, tool_format)` dataclass with `from_spec()` parser, proper `__hash__`/`__eq__` (for dict keys), and `__str__` (for display)
- **`run.py`**: `run_evals()` accepts and returns `dict[ModelConfig, ...]` instead of `dict[str, ...]`
- **`main.py`**: 
  - Model spec parsing builds `ModelConfig` objects directly
  - CSV output has separate "Model" and "Tool Format" columns
  - Display tables show model and format in distinct columns  
  - Directory layout: `model/tool_format/test_name` (was `model@format/test`)
  - Backward-compatible CSV reading handles legacy single-column format

### Why

The old `f"{model}@{tool_format}"` pattern:
- Lost type safety (string -> parse -> string round-trip)
- Was ambiguous with OpenRouter model names containing `@` (e.g. `z-ai/glm-5@z-ai`)
- Required repeated parsing of the `@` separator
- Mixed model identity with tool format in dict keys and file paths

### Test plan

- [x] All existing eval tests pass (`test_eval_cli`, `test_eval[hello-patch]`)
- [x] mypy clean on all changed files
- [x] ruff lint + format clean
- [x] ModelConfig edge cases: hash equality, OpenRouter `@` in model names, `from_spec()` parsing
- [x] Backward-compatible CSV reading (legacy format without "Tool Format" column)

----

> [!IMPORTANT]
> Refactor eval system to use `ModelConfig` dataclass instead of `model@format` strings, improving type safety and clarity.
> 
>   - **Behavior**:
>     - Replaces `model@format` string pattern with `ModelConfig` dataclass in `main.py`, `run.py`, and `types.py`.
>     - `run_evals()` in `run.py` now uses `ModelConfig` instead of string keys.
>     - CSV output in `main.py` now has separate "Model" and "Tool Format" columns.
>     - Directory layout updated to `model/tool_format/test_name`.
>     - Backward-compatible CSV reading for legacy format.
>   - **Models**:
>     - Adds `ModelConfig` dataclass in `types.py` with `from_spec()` method for parsing.
>   - **Misc**:
>     - Updates to logging and error handling in `run.py` and `main.py`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=gptme%2Fgptme&utm_source=github&utm_medium=referral)<sup> for ae8a32108c8bbc16f9d04e693e0e04e5db917105. You can [customize](https://app.ellipsis.dev/gptme/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: gptme/eval/main.py [hunk 1]
```diff
@@ -484,14 +478,32 @@ def read_log_file(file_path: Path) -> str:
...
-            model = row["Model"]
-            test_dir = results_dir / model / row["Test"]
+            model_name = row["Model"]
+            tool_format = row.get("Tool Format", "")
+
+            if tool_format:
+                # New format: separate columns
+                config = ModelConfig(
+                    model=model_name, tool_format=cast(ToolFormat, tool_format)
+                )
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/eval/main.py:499]: Already handled — the CSV reader validates `tool_format` against `get_args(ToolFormat)` and logs a warning + skips the row if the value is unknown (main.py:492-496). The `cast` only happens after that validation passes.

File: gptme/eval/types.py [hunk 2]
```diff
@@ -1,17 +1,49 @@
...
+        If the spec contains '@' and the suffix is a valid ToolFormat,
+        it's treated as a format separator. Otherwise '@' is part of the
+        model name (e.g. OpenRouter 'z-ai/glm-5@z-ai').
+        """
+        if "@" in spec:
+            model, fmt = spec.rsplit("@", 1)
+            if fmt in get_args(ToolFormat):
+                return cls(model=model, tool_format=cast(ToolFormat, fmt))
+            # '@' was part of model name, not a format separator
+        if default_format:
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/eval/types.py]: Both of these are already addressed in the PR: `if default_format is not None:` (types.py:42) and validation against `get_args(ToolFormat)` before `cast` with a warning log if invalid (main.py:492-496).

[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/eval/types.py]: Fixed in 9bce5b0 — changed to `is not None`.

File: gptme/eval/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply gptme/eval/main.py:499]: Fixed in 9bce5b0 — added `get_args(ToolFormat)` validation with warning log before the `cast`.

**Comment by ErikBjare** [comment]
Merging
